### PR TITLE
[MDS-4804] Upload documents any time mine incidents v2

### PR DESCRIPTION
**Pull request by areyeslo**
## Objective 

[MDS-4804](https://bcmines.atlassian.net/browse/MDS-4804)
[MDS-4799](https://bcmines.atlassian.net/browse/MDS-4799)

- Ability to add documents and final report (in case of Dangerous Occurrence) in any stage except draft:
image

image

- Fix bug in document persistence:
[MDS-4799: Update Incident - Adding new documents to existing document collection removes "some" persisted documents](https://bcmines.atlassian.net/browse/MDS-4799)

### CODE REVIEW

File: services/minespace-web/src/components/Forms/incidents/IncidentForm.js [hunk 1]
```diff
@@ -672,156 +692,166 @@ const renderUploadInitialNotificationDocuments = (
...
-        <Col span={24}>
-          {formDisabled && (
-            <Typography.Title level={3} id="documentation">
-              {title}
-            </Typography.Title>
-          )}
-          <Row>
-            <Col xs={24} md={12}>
-              <Typography.Title level={4}>{subTitle}</Typography.Title>
+      {(!props.formValues?.documents || props.formValues?.documents?.length === 0) &&
```
Comment (BenHarrington) [inline services/minespace-web/src/components/Forms/incidents/IncidentForm.js]: Could we consolidate these conditional checks into named variables? I hate to be that guy, but trying to troubleshoot this later could be a bit tricky with how complicated this is. Give me a shout if you want to talk through it.

File: services/minespace-web/src/components/Forms/incidents/IncidentForm.js [hunk 2]
```diff
@@ -658,10 +658,30 @@ const renderUploadInitialNotificationDocuments = (
...
+    : [];
+
+  const formValuesInitialNotificationDocs = props.formValues?.initial_notification_documents
+    ? props.formValues?.initial_notification_documents?.filter(
+        (doc) => doc.mine_incident_document_type_code === "INI"
+      )
+    : [];
+
+  const initialDocumentsForm =
+    // initialIncidentDocumentsForm.length === 0?
```
Comment (BenHarrington) [inline services/minespace-web/src/components/Forms/incidents/IncidentForm.js]: Remove.

**Review by JustinMacaulay — COMMENTED** [review]
Looks good only thing is that it looks like the nav bar on the left doesnt work based on your screenshots.

File: services/minespace-web/src/components/Forms/incidents/IncidentForm.js [hunk 3]
```diff
@@ -670,158 +688,171 @@ const renderUploadInitialNotificationDocuments = (
     props.formValues?.mine_determination_type_code ||
     props.formValues?.determination_type_code === "DO";
 
+  const noDocumentsInForm =
+    !props.formValues?.documents || props.formValues?.documents?.length === 0;
```
Comment (BenHarrington) [inline services/minespace-web/src/components/Forms/incidents/IncidentForm.js]: Should documents ever be null or undefined?

Comment (areyeslo) [reply services/minespace-web/src/components/Forms/incidents/IncidentForm.js]: either you have not upload any document in the new incident or  the existing incident does not have any document

**Review by BenHarrington — DISMISSED** [review]
Looks good, thanks Arturo!

**Review by BenHarrington — APPROVED** [review]
Great work Arturo. Thanks for getting this component tidied up.

**Pull request merged**